codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttEncoder.java

```diff
@@ -388,7 +388,8 @@ private static ByteBuf encodeUnsubAckMessage(
                     message.idAndPropertiesVariableHeader().properties());
             try {
                 int variableHeaderBufferSize = 2 + propertiesBuf.readableBytes();
-                int payloadBufferSize = message.payload().unsubscribeReasonCodes().size();
+                int payloadBufferSize =
+                        message.payload() == null ? 0 : message.payload().unsubscribeReasonCodes().size();
```

Comment:
nit: consider storing `payload()` in a local var and use it. There are multiple calls to `message.payload()`